supervision/detection/core.py

```diff
@@ -421,22 +428,25 @@ def from_transformers(cls, transformers_results: dict) -> Detections:
             ```
         """  # noqa: E501 // docs
 
+        class_ids = transformers_results["labels"].cpu().detach().numpy().astype(int)
+        data = {}
+        if id2label is not None:
+            class_names = np.array([id2label[class_id] for class_id in class_ids])
+            data[CLASS_NAME_DATA_FIELD] = class_names
         if "boxes" in transformers_results:
             return cls(
                 xyxy=transformers_results["boxes"].cpu().detach().numpy(),
                 confidence=transformers_results["scores"].cpu().detach().numpy(),
-                class_id=transformers_results["labels"]
-                .cpu()
-                .detach()
-                .numpy()
-                .astype(int),
+                class_id=class_ids,
+                data=data,
             )
         masks = transformers_results["masks"].cpu().detach().numpy().astype(bool)
```

Comment:
While we're at it, let's make the code more defensive.

I've reviewed the [image_processing_detr.py](https://github.com/huggingface/transformers/blob/fed27ffc7ec62837dca9bbfc83442eb3678ee026/src/transformers/models/detr/image_processing_detr.py#L744) in `transformers` and it turns out, with instance or panoptic segmentation `transformers_results` might not have `boxes`, `masks` or `labels`.

Let's add an `elif "masks" in transformers_results:` here and at the end - `raise NotImplementedError("Only object detection and semantic segmentation results are supported")`.